src/cpp/src/sampler.cpp

```diff
@@ -1036,7 +1037,7 @@ void Sampler::GroupBeamSearcher::Group::is_done() {
         return;
     }
     case ov::genai::StopCriteria::NEVER: {
-        size_t length = sampling_params.length_penalty > 0.0 ? sequence_group->get_max_new_tokens() : cur_len;
+        size_t length = sampling_params.length_penalty > 0.0 ? sequence_group->get_max_new_tokens(this->prompt_len) : cur_len;
```

Comment:
`sequence_group->get_max_new_tokens()` already uses `prompt_len`. I think this should be reverted